### PR TITLE
Test book

### PR DESCRIPTION
**Pull request by nicoopalenque**
# Sumary

* Create tests for Index and Show routes from Book

# Card

[https://trello.com/c/QG6iF4CF](url)

# Screenshot

Captura de Pantalla 2021-06-14 a la(s) 15 08 25

### CODE REVIEW

File: spec/controller/book_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::BookController, type: :controller do
+  include_context 'User Auth'
+  describe 'GET #index' do
```
Comment (blacksam07) [inline spec/controller/book_controller_spec.rb]: use fuctium

File: spec/controller/book_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    context 'When fetching all books' do
+      let!(:book) { create_list(:book, 3) }
+
+      before do
+        get :index
+      end
+
+      it 'response with books json' do
+        resp = JSON.parse(response.body)
```
Comment (blacksam07) [inline spec/controller/book_controller_spec.rb]: you can use be_paginated of wor-paginate https://github.com/Wolox/wor-paginate#test-helpers

File: spec/factories/user.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FactoryBot.define do
+  factory :user do
+    email { 'test@wolox.com.ar' }
```
Comment (blacksam07) [inline spec/factories/user.rb]: Use Faker

**Review by blacksam07 — CHANGES_REQUESTED** [review]

File: spec/controller/book_controller_spec.rb [hunk 4]
```diff
@@ -1,39 +1,38 @@
...
-        books = []
-        book.each do |item|
-          books << Api::V1::Serializer::BookSerializer.new.serialize_to_json(item)
-        end
-        books.each_with_index do |item, index|
-          expect(JSON.parse(item)).to eq(resp['page'][index])
-        end
+      let(:response_body) do
+        ActiveSupport::JSON.decode(response.body) if response.present? && response.body.present?
+      end
```
Comment (blacksam07) [inline spec/controller/book_controller_spec.rb]: you can define this inside spec/support/parsed_response_body.rb

```ruby
module Response
  module JSONParser
    def response_body
      ActiveSupport::JSON.decode(response.body) if response.present? && response.body.present?
    end
  end
end
```

**Review by blacksam07 — APPROVED** [review]

File: spec/controller/book_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::BookController, type: :controller do
```
Comment (redwarewolf) [inline spec/controller/book_controller_spec.rb:3]: You are missing the tests for failure scenarios

**Review by redwarewolf — CHANGES_REQUESTED** [review]

**Pull request merged**